cmd/goal/clerk.go

```diff
@@ -344,18 +349,14 @@ var sendCmd = &cobra.Command{
 					Args:  programArgs,
 				},
 			}
-		} else if sign || txFilename == "" {
-			wh, pw := ensureWalletHandleMaybePassword(dataDir, walletName, true)
-			stx, err = client.SignTransactionWithWallet(wh, pw, payment)
+		} else {
+			signTx := sign || (txFilename == "")
+			stx, err = createSignedTransaction(client, signTx, dataDir, walletName, payment)
 			if err != nil {
 				reportErrorf(errorSigningTX, err)
 			}
-		} else {
-			stx = transactions.SignedTxn{
-				Txn: payment,
-			}
-			stx = populateBlankMultisig(client, dataDir, walletName, stx)
 		}
+
 		if txFilename == "" {
 			// Sign and broadcast the tx
```

Comment:
update this comment to reflect that the tx will have already been signed

Reply:
done